packages/react-native-analytics/src/integrations/castle/Castle.js

```diff
@@ -0,0 +1,229 @@
+      );
+
+      return;
+    }
+
+    if (configureHttpClientHeaderCustomFn) {
+      try {
+        await configureHttpClientHeaderCustomFn(this);
+
+        this.isInterceptorAttached = true;
+      } catch (error) {
+        utils.logger.error(
+          `${LOGGER_MESSAGE_PREFIX} There was an error trying to execute the "configureHttpClientHeader" custom function. ${error}`,
+        );
+
+        this.isInterceptorAttached = false;
+      }
+
+      return;
+    }
+
+    // Default configuration of our @farfetch/blackout-core client (axios) using an interceptor.
+    // Store the interceptor on the instance in case the user wants to remove it.
+    this.httpClientInterceptor = this.httpClient?.interceptors?.request?.use(
+      this.onBeforeRequestFullfill,
+      this.onBeforeRequestReject,
+    );
+  }
+
+  /**
+   * Callback that is used on the Axios interceptor to add the correct Castle.io token header.
+   *
+   * @async
+   *
+   * @param {object} config - Axios config object.
+   *
+   * @returns {object} - The modified Axios config object.
+   */
+  onBeforeRequestFullfill = async config => {
+    await this.initializePromise;
+
+    const tokenHeaderName = await this.castleIO.requestTokenHeaderName();
+    const requestTokenValue = await this.castleIO.createRequestToken();
+
+    config.headers[tokenHeaderName] = requestTokenValue;
+
+    this.isInterceptorAttached = true;
+
+    return config;
+  };
+
+  /**
+   * Callback that is used on the Axios interceptor to handle any error when setting the Castle.io header.
+   *
+   * @async
+   *
+   * @param {object|string} error - The error itself.
+   */
+  onBeforeRequestReject = async error => {
+    this.isInterceptorAttached = false;
+
+    utils.logger.error(
+      `${LOGGER_MESSAGE_PREFIX} Castle integration failed to append the necessary header.`,
+    );
+
+    return Promise.reject(error);
+  };
+
+  /**
+   * Handles when the onSetUser is called on the analytics side. Logs when a login occurs and sets both the ID and user traits.
+   *
+   * @async
+   *
+   * @param {object} data - Event data provided by analytics.
+   */
+  async onSetUser(data) {
+    await this.initializePromise;
+
+    if (!this.httpClient || !this.isInterceptorAttached) {
+      utils.logger.error(HTTP_CLIENT_ERROR_MESSAGE);
+
+      return;
+    }
+
+    const userData = data?.user || {};
+    const userId = userData.id;
+    const traits = userData.traits;
+    const isGuest = traits?.isGuest;
+
+    // If for some reason there was a call to `analytics.onSetUser()` that receives the same user id, ignore it.
+    if (userId === this.lastUserId) {
+      return;
+    }
+
+    try {
+      // Login - Let Castle identify the user.
+      if (userId && !isGuest) {
+        await this.castleIO.identify(userId, traits);
+      } else {
+        // Logout
+        await this.castleIO.reset();
+      }
+
+      this.lastUserId = userId;
```

Comment:
This instruction should be added after the try/catch block to avoid an error when calling `this.castleIO.identify` or `this.castleIO.reset` preventing the update of the lastUserId member variable.

Reply:
fixed 🙂  thanks!